R/tableOmopSnapshot.R

```diff
@@ -0,0 +1,68 @@
+#' Create a gt table from a summarised omop snapshot
+#'
+#' @param omopSnapshot  Output from summariseOmopSnapshot()
+#'
+#' @return A gt object with the summarised data.
+#' @export
+#'
+#' @examples
+#' \donttest{
+#'library(dplyr)
+#'library(OmopSketch)
+#'
+#' cdm <- mockOmopSketch(numberIndividuals = 1000)
+#'
+#' cdm |>
+#' summariseOmopSnapshot() |>
+#' tableOmopSnapshot()
+#'}
+tableOmopSnapshot <- function(summarisedOmopSnapshot){
+  # Initial checks ----
+  omopgenerics::assertClass(summarisedOmopSnapshot, "summarised_result")
+
+  if(summarisedOmopSnapshot |> dplyr::tally() |> dplyr::pull("n") == 0){
+    cli::cli_warn("summarisedOmopSnapshot is empty.")
+
+    return(
+      summarisedOmopSnapshot |>
+        visOmopResults::splitGroup() |>
+        dplyr::select("Estimate" = "variable_name", "cdm_name") |>
+        gt::gt()
+    )
+  }
+
+  t <- dplyr::tibble(
```

Comment:
why is this needed?

Reply:
If you visualise the plot you may understand it

Reply:
Here is a plot of the outcome:
<img width="1174" alt="Screenshot 2024-09-16 at 21 40 02" src="https://github.com/user-attachments/assets/6ab38920-f7b6-4842-9661-bf89da7b5efe">

Reply:
And here a plot of the summarised result:
<img width="1055" alt="Screenshot 2024-09-16 at 21 41 01" src="https://github.com/user-attachments/assets/8deb0ed5-1180-441c-b8d7-1d3e6493c296">